### PR TITLE
feat(CMake): enable optional C++20 support

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Add new cmake option `USE_CPP_20` for use new `c++20` standard

### CODE REVIEW

File: conf/dist/config.cmake [hunk 1]
```diff
@@ -84,3 +84,6 @@ endif()
 # Source tree in IDE
 set(WITH_SOURCE_TREE       "hierarchical" CACHE STRING "Build the source tree for IDE's.")
 set_property(CACHE WITH_SOURCE_TREE PROPERTY STRINGS no flat hierarchical)
+
+# If disable - use c++17
+option(USE_CPP_20 "Enable c++20 standart" 0)
```
Comment (FrancescoBorzi) [inline conf/dist/config.cmake:89]: missing new line

Comment (Winfidonarleyan) [reply conf/dist/config.cmake]: Fix

File: src/cmake/macros/ConfigureBaseTargets.cmake [hunk 2]
```diff
@@ -9,39 +9,26 @@ add_library(acore-compile-option-interface INTERFACE)
...
-    cxx_trailing_return_types
-    cxx_return_type_deduction)
+if (USE_CPP_20)
+  # Enable support С++20
+  set(CMAKE_CXX_STANDARD 20)
+  message(STATUS "Enabled С++20 standart")
+else()
+  # Enable support С++17
+  set(CMAKE_CXX_STANDARD 17)
+  message(STATUS "Enabled С++17 standart")
```
Comment (Kitzunu) [inline src/cmake/macros/ConfigureBaseTargets.cmake:19]:
```suggestion
  message(STATUS "Enabled С++20 standard")
else()
  # Enable support С++17
  set(CMAKE_CXX_STANDARD 17)
  message(STATUS "Enabled С++17 standard")
```

File: conf/dist/config.cmake (hunk 1 above)
Comment (Kitzunu) [inline conf/dist/config.cmake:89]:
```suggestion
option(USE_CPP_20 "Enable c++20 standard" 0)
```

**Comment by Winfidonarleyan** [comment]
@Kitzunu thanks for correct

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**